File: ios/AstrovoxAI/Services/APIService.swift
import Foundation

class APIService: ObservableObject {
    static let shared = APIService()
    private let baseURL = "https://api.astrovox.ai/v1"
    private let session = URLSession.shared

    func sendMessage(conversationId: UUID, message: String, model: String = "gpt-4") async throws -> [String: Any] {
        let url = URL(string: "\(baseURL)/chat/message")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        let body = ["conversation_id": conversationId.uuidString, "message": message, "model": model]
        request.httpBody = try JSONSerialization.data(withJSONObject: body)
        let (data, _) = try await session.data(for: request)
        return try JSONSerialization.jsonObject(with: data) as! [String: Any]
    }

    func healthCheck() async throws -> [String: Any] {
        let url = URL(string: "\(baseURL)/health")!
        let (data, _) = try await session.data(from: url)
        return try JSONSerialization.jsonObject(with: data) as! [String: Any]
    }
}

File: android/app/src/main/java/ai/astrovox/data/remote/ApiService.kt
package ai.astrovox.data.remote

import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

data class ChatRequest(
    val conversation_id: String,
    val message: String,
    val model: String
)

interface AstrovoxApi {
    @POST("/v1/chat/message")
    suspend fun sendMessage(@Body request: ChatRequest): Map<String, Any>

    @GET("/v1/health")
    suspend fun healthCheck(): Map<String, Any>
}

object ApiService {
    private val retrofit = RetrofitClient.retrofit
    fun create(): AstrovoxApi = retrofit.create(AstrovoxApi::class.java)
}
